crates/tracing/src/tracing/mod.rs

```diff
@@ -0,0 +1,173 @@
+        })
+        .init()
+        .map_err(|e| format!("{:?}", e))?;
+    let provider = sdk::Provider::builder()
+        .with_simple_exporter(exporter)
+        .with_config(sdk::Config {
+            default_sampler: Box::new(sdk::Sampler::Always),
+            ..Default::default()
+        })
+        .build();
+    let tracer = provider.get_tracer("tracing");
+    global::set_provider(provider);
+    let jaeger_layer = JaegerLayer::new(tracer);
+    Ok(jaeger_layer)
+}
+
+pub fn span_context(span: &tracing::Span) -> Option<crate::SpanContext> {
+    tracing::dispatcher::get_default(|dispatch| {
+        dispatch
+            .downcast_ref::<Registry>()
+            .or_else(|| {
+                dbg!("Failed to downcast dispatch");
+                None
+            })
+            .and_then(|s| {
+                s.span(&span.id().expect("Failed to get span id"))
+                    .and_then(|span_ref| {
+                        span_ref
+                            .extensions()
+                            .get::<api::SpanContext>()
+                            .or_else(|| {
+                                dbg!("Context was already in span");
+                                None
+                            })
+                            .cloned()
+                    })
+            })
+    })
+    .or_else(|| {
+        dbg!("Failed to get dispatch");
+        None
+    })
+    .and_then(|context| {
+        let builder = rustracing_jaeger::span::SpanContextStateBuilder::new();
+        let c = context.trace_id();
+        format!("{:x}", c)
+            .parse::<rustracing_jaeger::span::TraceId>()
+            .ok()
+            .map(|trace_id| {
+                builder
+                    .trace_id(trace_id)
+                    .span_id(context.span_id())
+                    .finish()
+            })
+    })
+    .map(|context| {
+        let context = rustracing::span::SpanContext::new(context, vec![]);
+        crate::span_context::HSpanContext(context)
+    })
+}
+
+/*
+pub fn span_context(id: &Id) -> Option<crate::SpanContext> {
+    tracing::dispatcher::get_default(|d| {
+        d.is::<Layered<JaegerLayer, Registry>>();
+        d.downcast_ref::<Layered<JaegerLayer, Registry>>()
+            .and_then(|s| s.span(&id))
+            .and_then(|s| s.extensions().get::<BoxedSpan>().map(|s| s.get_context()))
+    })
+    .and_then(|context| {
+        let builder = rustracing_jaeger::span::SpanContextStateBuilder::new();
+        let c = context.trace_id();
+        format!("{:x}", c)
+            .parse::<rustracing_jaeger::span::TraceId>()
+            .ok()
+            .map(|trace_id| {
+                builder
+                    .trace_id(trace_id)
+                    .span_id(context.span_id())
+                    .finish()
+            })
+    })
+    .map(|context| {
+        let context = rustracing::span::SpanContext::new(context, vec![]);
+        crate::span_context::HSpanContext(context)
+    })
+}
+*/
```

Comment:
There's a lot of commented out sections in this file. Suggest deleting or adding comments why we're keeping it.

Reply:
These are all behind a compile flag that is disabled so they won't even make it into the code or docs. I just didn't want to loose the ideas if we come back to this

Reply:
happy to rip it into a gist though

Reply:
I'm also totally cool leaving them here if they're useful : )